cmd/loadtest/loadtest.go

```diff
@@ -963,9 +970,32 @@ func loadtestTransaction(ctx context.Context, c *ethclient.Client, nonce uint64)
 	chainID := new(big.Int).SetUint64(*ltp.ChainID)
 	privateKey := ltp.ECDSAPrivateKey
 
+	head, err := c.HeaderByNumber(ctx, nil)
+	if err != nil {
+		log.Error().Err(err).Msg("Unable to get head")
+		return
+	}
+
 	gasLimit := uint64(21000)
-	tx := ethtypes.NewTransaction(nonce, *to, amount, gasLimit, gasPrice, nil)
-	stx, err := ethtypes.SignTx(tx, ethtypes.NewEIP155Signer(chainID), privateKey)
+	var tx *ethtypes.Transaction
+	if *ltp.LegacyTransactionMode {
+		tx = ethtypes.NewTransaction(nonce, *to, amount, gasLimit, gasPrice, nil)
+	} else {
+		gasTipCap := ltp.CurrentGasTipCap
+		gasFeeCap := new(big.Int).Add(gasTipCap, new(big.Int).Mul(head.BaseFee, big.NewInt(2)))
```

Comment:
There is some panic happening here in testing unfortunately. Even without the panic, I'm unsure what this logic is actually doing. The logic seems to be `maxFeePerGas = masPriorityFeePerGas + 2*BaseFee` but why?

![image](https://github.com/maticnetwork/polygon-cli/assets/429588/92a804b3-bc15-412f-b8c8-299ecb4ad837)